Set the default value of "VIMRUNTIME" to `$EXEDIR/vimruntime` where `$EXEDIR`
is the directory in which the Vim binary resides.

diff --git src/misc1.c src/misc1.c
index b27c527df..1b9d8a050 100644
--- src/misc1.c
+++ src/misc1.c
@@ -24,6 +24,9 @@
 // All user names (for ~user completion as done by shell).
 static garray_T	ga_users;
 
+#include <string.h>
+#include "../../exepath.c"
+
 /*
  * get_leader_len() returns the length in bytes of the prefix of the given
  * string which introduces a comment.  If this string is not a comment then
@@ -1629,6 +1632,7 @@ vim_getenv(char_u *name, int *mustfree)
     char_u	*p = NULL;
     char_u	*pend;
     int		vimruntime;
+    char_u	path[PATH_MAX + 10]; // 10 == strlen("vimruntime");
 #ifdef MSWIN
     WCHAR	*wn, *wp;
 
@@ -1670,6 +1674,14 @@ vim_getenv(char_u *name, int *mustfree)
     if (!vimruntime && STRCMP(name, "VIM") != 0)
 	return NULL;
 
+	if (vimruntime && !p && executable_path(path)) {
+	    *mustfree = TRUE;
+	    STRCPY(rindex(path, '/') + 1, "vimruntime");
+	    vim_setenv((char_u *) "VIMRUNTIME", path);
+	    didset_vimruntime = 1;
+	    return strdup(path);
+	}
+
     /*
      * When expanding $VIMRUNTIME fails, try using $VIM/vim<version> or $VIM.
      * Don't do this when default_vimruntime_dir is non-empty.
